File: app/src/main/java/com/example/daisyling/common/util/download/DownloadInterceptor.java
package com.example.daisyling.common.util.download;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Response;

/**
 * Created by Emily on 10/11/21
 */
public class DownloadInterceptor implements Interceptor {
    private DownloadListener listener;

    public DownloadInterceptor(DownloadListener listener) {
        this.listener = listener;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Response originalResponse = chain.proceed(chain.request());

        return originalResponse.newBuilder()
                .body(new DownloadResponseBody(originalResponse.body(), listener))
                .build();
    }
}

File: app/src/main/java/com/example/daisyling/common/util/download/DownloadInterceptor.kt
package com.example.daisyling.common.util.download

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

/**
 * Created by Emily on 10/11/21
 */
class DownloadInterceptor(private val listener: DownloadListener) : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val originalResponse: Response = chain.proceed(chain.request())
        return originalResponse.newBuilder()
            .body(DownloadResponseBody(originalResponse.body!!, listener))
            .build()
    }
}